dotnet/src/SemanticKernel.UnitTests/TemplateEngine/AggregatorPromptTemplateFactoryTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.TemplateEngine;
+using Xunit;
+
+namespace SemanticKernel.UnitTests.TemplateEngine;
+public sealed class AggregatorPromptTemplateFactoryTests
+{
+    [Fact]
+    public void ItCreatesMyPromptTemplates()
+    {
+        // Arrange
+        var templateString = "{{$input}}";
+        var target = new AggregatorPromptTemplateFactory(new MyPromptTemplateFactory1(), new MyPromptTemplateFactory2());
+
+        // Act
+        var result1 = target.CreatePromptTemplate(templateString, new PromptTemplateConfig() { TemplateFormat = "my-format-1" });
+        var result2 = target.CreatePromptTemplate(templateString, new PromptTemplateConfig() { TemplateFormat = "my-format-2" });
+
+        // Assert
+        Assert.NotNull(result1);
+        Assert.True(result1 is MyPromptTemplate1);
+        Assert.NotNull(result2);
+        Assert.True(result2 is MyPromptTemplate2);
+    }
+
+    [Fact]
+    public void ItThrowsExceptionForUnknowPromptTemplateFormat()
+    {
+        // Arrange
+        var templateString = "{{$input}}";
+        var target = new AggregatorPromptTemplateFactory(new MyPromptTemplateFactory1(), new MyPromptTemplateFactory2());
+
+        // Act
+        var result1 = target.CreatePromptTemplate(templateString, new PromptTemplateConfig() { TemplateFormat = "my-format-1" });
+        var result2 = target.CreatePromptTemplate(templateString, new PromptTemplateConfig() { TemplateFormat = "my-format-2" });
+
+        // Assert
+        Assert.Throws<SKException>(() => target.CreatePromptTemplate(templateString, new PromptTemplateConfig() { TemplateFormat = "unknown-format" }));
+    }
+}
+
+public class MyPromptTemplateFactory1 : IPromptTemplateFactory
```

Comment:
If those classes are only used in this test, change them as inner classes, otherwise we should move those `Fakes` to a Fakes folder.

Reply:
Done